crates/wit/README.md

```diff
@@ -0,0 +1,204 @@
+# The `wit` tool
+
+A tool for creating and publishing WIT packages to a [WebAssembly component
+registry](https://warg.io/).
+
+WIT packages are used in the [WebAssembly Component Model](https://github.com/WebAssembly/component-model/)
+for defining interfaces and types used in WebAssembly components.
+
+## Requirements
+
+* The `wit` tool is written in Rust, so you'll want the [latest stable Rust
+  installed](https://www.rust-lang.org/tools/install).
+
+## Installation
+
+```
+cargo install --bin wit --locked --git https://github.com/bytecodealliance/cargo-component
+```
+
+## Initializing a WIT package
+
+To initialize a new WIT package in the current directory:
+
+```
+wit init
+```
+
+This creates a `wit.toml` file with the following contents:
+
+```toml
+version = "0.1.0"
+
+[dependencies]
+
+[registries]
+```
+
+By default, the WIT package will not have any dependencies specified.
+
+The registries section contains a mapping of registry names to URLs.
+
+A registry named `default` will be the registry to use when a dependency does
+not explicitly specify a registry.
```

Comment:
It may be worth mentioning that the reason we need this is that the design is for federated registries, rather than having a single central registry that gets used by default.